src/firefly/java/edu/caltech/ipac/firefly/messaging/Messenger.java

```diff
@@ -73,7 +77,7 @@ static boolean init() {
     public static String getStats() {
 
         JsonHelper stats = new JsonHelper();
-        if (!init()) {
+        if (connectAttempt!=ConnectAttempt.CONNECTED || jedisPool==null) {
             return stats.setValue("Messenger is offline").toJson();
```

Comment:
`init()` is needed before every call.  It's a lazy-load approach to ensure it's called before it's used.
Also, since you added status, you can do something like this:
```
        init();

        stats.setValue(status, "status");
        if (status == Status.CONNECTED) {
            ....
        }
        return stas.toJson();
```